### PR TITLE
[3.x] Sharing `repeat` iteration as `dataset` variable

### PR DESCRIPTION
**Pull request by Katalam**
### What:

- [ ] Bug Fix
- [ ] New Feature
- [x] Improvement

### Description:

- Essentially, what was previously in [Issue #908](https://github.com/pestphp/pest/issues/908) requested
- When no dataset is provided, but the repetitions are above 1, the current iteration should be passed to the closure of the test

### Related:
[Issue #908](https://github.com/pestphp/pest/issues/908)

### CODE REVIEW

**Review by devajmeireles — APPROVED** [review]

**Comment by devajmeireles** [comment]
Hey, @Katalam ! Thanks for your PR.

Everything works as expected. Approved. 👍🏻

File: tests/Features/Repeat.php [hunk 1]
```diff
@@ -16,3 +16,7 @@
     expect([1, 2, 3])->toContain($numberA)
         ->and([4, 5, 6])->toContain($numberB);
 })->repeat(times: 7)->with(['a' => 1, 'b' => 2, 'c' => 3], [4, 5, 6]);
+
+test('multiple times with iterator as argument', function (int $iteration) {
+    expect($iteration)->toBeGreaterThan(0);
+})->repeat(times: 8);
```
Comment (utsavsomaiya) [inline tests/Features/Repeat.php]: Can you please write a test with the dataset so we can get both as closure arguments!!

**Comment by Katalam** [comment]
@devajmeireles hey, should the conflict between repeat and dataset at the same time be solved by passing both iteration variables in the closure? I can do that

File: tests/Features/Repeat.php (hunk 1 above)
Comment (devajmeireles) [reply tests/Features/Repeat.php]: > Can you please write a test with the dataset so we can get both as closure arguments!!

@Katalam , could you write a test that uses the `repeat` and `with` to validate if all will work as expected, please?

Comment (Katalam) [reply tests/Features/Repeat.php]: @devajmeireles Done

**Review by devajmeireles — APPROVED** [review]

**Pull request merged**